src/python/zquantum/optimizers/daemon_optimizer/optimize/_api.py

```diff
@@ -1,10 +1,8 @@
 from zquantum.core.circuit import save_circuit_template_params
 from zquantum.core.utils import load_value_estimate, convert_array_to_dict, SCHEMA_VERSION
-
+from .cost_function import ProxyCostFunction
 import scipy
-import time
 import json
-import io
 
 # ---------- Optimize Variational Circuit ----------
 def optimize_variational_circuit_with_proxy(initial_params, optimizer, client, **kwargs):
```

Comment:
I think we might be able to collapse this into the `optimize_variational_circuit` function by making the initial and final status calls to the proxy in the template.

Reply:
Apologies -> meant collapse this into intializing the cost function and calling optimizer.minimize in the template